inference-engine/src/transformations/src/transformations/common_optimizations/shuffle_channels_fusion.cpp

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/shuffle_channels_fusion.hpp"
+#include "itt.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+bool check_shapes(const ngraph::Shape& shape_input, const ngraph::Shape& shape_reshape_before,
+                  const ngraph::AxisVector& transpose_constant_values, const ngraph::Shape& shape_reshape_after) {
+    // x: [N, C, H, W]
+    bool is_transformation_valid = (shape_input.size() == 4);
+
+    // x'= reshape(x, [N, group, C / group, H * W]) or reshape(x, [N, group, C / group, H, W])
+    bool is_reshape_before_valid = (shape_reshape_before.size() == 4 || shape_reshape_before.size() == 5);
+    if (is_reshape_before_valid) {
+        size_t group = shape_reshape_before[1];
+        ngraph::Shape expected_reshape_before = { shape_input[0], group, shape_input[1] / group };
+
+        if (shape_reshape_before.size() == 4) {
+            expected_reshape_before.push_back(shape_input[2] * shape_input[3]);
+        } else {
+            expected_reshape_before.push_back(shape_input[2]);
+            expected_reshape_before.push_back(shape_input[3]);
+        }
+
+        is_reshape_before_valid &= (expected_reshape_before == shape_reshape_before);
+    }
+
+    // x''= transpose(x', [0, 2, 1, 3]) or transpose(x', [0, 2, 1, 3, 4])
+    bool is_transpose_valid = (transpose_constant_values.size() == 4 || transpose_constant_values.size() == 5);
+    if (is_transpose_valid) {
+        ngraph::AxisVector expected_transpose_values{ 0, 2, 1, 3 };
+        if (transpose_constant_values.size() == 5) {
+            expected_transpose_values.push_back(4);
+        }
+
+        is_transpose_valid &= (expected_transpose_values == transpose_constant_values);
+    }
+
+    // y = reshape(x'', [N, C, H, W])
+    bool is_reshape_after_valid = (shape_input == shape_reshape_after);
+
+    is_transformation_valid &= is_reshape_before_valid & is_transpose_valid & is_reshape_after_valid;
+    return is_transformation_valid;
+}
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ShuffleChannelsFusion, "ShuffleChannelsFusion", 0);
+
+ngraph::pass::ShuffleChannelsFusion::ShuffleChannelsFusion(const bool reshape_constants_check) {
+    MATCHER_SCOPE(ShuffleChannelsFusion);
+    auto input = ngraph::pattern::any_input(pattern::has_static_shape());
+    auto reshape_before_const_pattern = ngraph::pattern::wrap_type<ngraph::opset6::Constant>(pattern::has_static_shape());
+    auto transpose_const_pattern = ngraph::pattern::wrap_type<ngraph::opset6::Constant>(pattern::has_static_shape());
+    auto reshape_after_const_pattern = ngraph::pattern::wrap_type<ngraph::opset6::Constant>(pattern::has_static_shape());
+
+    auto has_static_shape_and_single_consumer = [](const Output<Node>& output) {
+        return pattern::has_static_shape() && pattern::consumers_count(1);
+    };
```

Comment:
This predicates `pattern::has_static_shape` and `pattern::consumers_count(1)` return lambda function, so I assume that
 `&&` for two lambdas will always return true. You need to call this labmdas for an output like this:
```suggestion
    auto has_static_shape_and_single_consumer = [](const Output<Node>& output) {
        return pattern::has_static_shape()(output) && pattern::consumers_count(1)(output);
    };
```

Reply:
Thanks, fixed!